weldx/util.py

```diff
@@ -1201,3 +1193,78 @@ def time_ref(self, value: pd.Timestamp):
                 self._obj.time.attrs["time_ref"] = value  # set new time_ref value
             else:
                 self._obj.time.attrs["time_ref"] = value
+
+
+class _Eq_compare_nested:
+    """Compares nested data structures like lists, sets, tuples, arrays, etc."""
+
+    compare_funcs = {
+        (np.ndarray, pint.Quantity): lambda x, y: np.all(x == y),
```

Comment:
```suggestion
        (np.ndarray, pint.Quantity, pd.Index): lambda x, y: np.all(x == y),
```
this one should be pretty common as well